src/libstd/sys/unix/ext/net.rs

```diff
@@ -561,12 +483,19 @@ impl SocketAncillary {
     /// }
     /// ```
     #[unstable(feature = "unix_socket_ancillary_data", issue = "none")]
-    pub fn add_fds(&mut self, fds: &[RawFd]) {
+    pub fn add_fds(&mut self, fds: &[RawFd]) -> bool {
```

Comment:
The lifetime here would only ensure that the `RawFd` instances live long enough until they're copied into the buffer, not until they're handed over to the kernel when actually sending this ancillary data. Which means by the time the send is executed they could be invalid. That's probably not a safety issue but certainly an ergonomic one.

At minimum this would have to be noted in the method descriptions or we have to come up with a better lifetime story.

Reply:
Yes, you are right. I think it should be noted in the description. I am open to your suggestions.

Reply:
RawFd already requires manual management and cleanup of the descriptors, so adding it to the method descriptions is probably sufficient.